src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/services/service-form/service-form.component.ts

```diff
@@ -0,0 +1,334 @@
+        null,
+        [
+          CdValidators.requiredIf({
+            service_type: 'nfs',
+            unmanaged: false
+          }),
+          CdValidators.requiredIf({
+            service_type: 'iscsi',
+            unmanaged: false
+          })
+        ]
+      ],
+      // NFS
+      namespace: [null],
+      // RGW
+      rgw_frontend_port: [
+        null,
+        [CdValidators.number(false), Validators.min(1), Validators.max(65535)]
+      ],
+      // iSCSI
+      trusted_ip_list: [null],
+      api_port: [null, [CdValidators.number(false), Validators.min(1), Validators.max(65535)]],
+      api_user: [
+        null,
+        [
+          CdValidators.requiredIf({
+            service_type: 'iscsi',
+            unmanaged: false
+          })
+        ]
+      ],
+      api_password: [
+        null,
+        [
+          CdValidators.requiredIf({
+            service_type: 'iscsi',
+            unmanaged: false
+          })
+        ]
+      ],
+      // RGW & iSCSI
+      ssl: [false],
+      ssl_cert: [
+        '',
+        [
+          CdValidators.composeIf(
+            {
+              service_type: 'rgw',
+              unmanaged: false,
+              ssl: true
+            },
+            [Validators.required, CdValidators.sslCert()]
+          ),
+          CdValidators.composeIf(
+            {
+              service_type: 'iscsi',
+              unmanaged: false,
+              ssl: true
+            },
+            [Validators.required, CdValidators.sslCert()]
+          )
+        ]
+      ],
+      ssl_key: [
+        '',
+        [
+          CdValidators.composeIf(
+            {
+              service_type: 'rgw',
+              unmanaged: false,
+              ssl: true
+            },
+            [Validators.required, CdValidators.sslPrivKey()]
+          ),
+          CdValidators.composeIf(
+            {
+              service_type: 'iscsi',
+              unmanaged: false,
+              ssl: true
+            },
+            [Validators.required, CdValidators.sslPrivKey()]
+          )
+        ]
+      ]
+    });
+  }
+
+  ngOnInit(): void {
+    this.action = this.actionLabels.CREATE;
+    this.cephServiceService.getKnownTypes().subscribe((resp: Array<string>) => {
+      // Remove service type 'osd', this is deployed a different way.
+      this.serviceTypes = _.difference(resp, ['osd']).sort();
+    });
+    this.hostService.list().subscribe((resp: object[]) => {
+      const options: SelectOption[] = [];
+      _.forEach(resp, (host: object) => {
+        if (_.get(host, 'sources.orchestrator', false)) {
+          const option = new SelectOption(false, _.get(host, 'hostname'), '');
+          options.push(option);
+        }
+      });
+      this.hosts.options = [...options];
+    });
+    this.hostService.getLabels().subscribe((resp: string[]) => {
+      this.labels = resp;
+    });
+    this.poolService.getList().subscribe((resp: Array<object>) => {
+      this.pools = resp;
+    });
```

Comment:
(Optional) can be forkJoin.

Reply:
Calling `forkJoin` with only one source does not make any sense.